### ISSUE TITLE
Mapping of class extending generic class

### ISSUE THREAD

**Issue by beleta** [vojtechhabarta/typescript-generator#436]
I am trying to map the following class:

``` java
public class DictionaryEntry implements java.util.Map.Entry<String, String> {
	private String key;
	private String value;

	public DictionaryEntry() {
	}

	public DictionaryEntry(String key, String value) {
		this.setKey(key);
		this.value = value;
	}

	public String getEntryValue() {
		return value;
	}

	@Override
	@JsonIgnore
	public String getKey() {
		return key;
	}

	@Override
	@JsonIgnore
	public String getValue() {
		return value;
	}

	public void setEntryValue(String entryValue) {
		this.value = entryValue;
	}

	public void setKey(String key) {
		this.key = key;
	}

	@Override
	public String setValue(String value) {
		String oldValue = this.value;
		this.value = value;
		return oldValue;
	}
}

````

What I get is:

``` typescript
export interface DictionaryEntry extends Entry<string, string> {
}

export interface Entry<K, V> {
}

```

But I think that the correct TypeScript generated code has to be:

``` typescript
export interface DictionaryEntry {
    entryValue: string;
}

export interface Entry<K, V> {
}

```

Any help would be appreciated.

**Comment by vojtechhabarta** [comment]
Are you using Jackson? `Map.Entry` has special handling in Jackson and thus also in typescript-generator.
Let's say we have a class that uses `DictionaryEntry`:
``` java
public class DictionaryEntryUsage {
    public DictionaryEntry dictionaryEntry;

    public DictionaryEntryUsage(DictionaryEntry dictionaryEntry) {
        this.dictionaryEntry = dictionaryEntry;
    }
}
```
when I serialize it using following code:
``` java
new ObjectMapper().writeValueAsString(new DictionaryEntryUsage(new DictionaryEntry("key1", "value1"));
```
result is:
``` json
{
    "dictionaryEntry": {
        "key1": "value1"
    }
}
```
When I generate TypeScript definition for `DictionaryEntryUsage` I get following:
``` typescript
interface DictionaryEntryUsage {
    dictionaryEntry: { [index: string]: any };
}
```
So I think this case works well. Do you have different case?

**Comment by beleta** [comment]
Yes, I am using Jackson 2.

But I am passing an array of  `DictionaryEntry `in my rest api, so I would like to access the properties of  `DictionaryEntry`, and clearly `entryValue `is a property of `DictionaryEntry`, i do not understand why it is not generated.

If I comment out `implements  java.util.Map.Entry<String, String>` then things go well.

**Comment by vojtechhabarta** [comment]
Yes, if you remove `implements Map.Entry` then `DictionaryEntry` is treated as regular object by both Jackson and typescript-generator.

Another possibility is to use `@JsonFormat` like this:
``` java
@JsonFormat(shape = JsonFormat.Shape.OBJECT)
public class DictionaryEntry implements java.util.Map.Entry<String, String> {
    // ...
}
```

While I was trying to use your class in array I realized that typescript-generator didn't support the **default** Jackson format of `Map.Entry` when it is used in arrays, collections etc.
Hopefully I fixed this in dabf7d2cecdddbdb59bbe9d1359aa51fe7054a6e.

Following is description of the change by example, Java classes:
``` java
public class DictionaryEntry implements Map.Entry<String, String> {}

public class DictionaryEntryUsage {
    public DictionaryEntry[] dictionaryEntries;  // array
}
```
output from Jackson:
``` json
{
    "dictionaryEntries": [
        {
            "key1": "value1"
        }
    ]
}
```
generated TypeScript code was:
``` typescript
interface DictionaryEntryUsage {
    dictionaryEntries: DictionaryEntry[];  // wrong
}
interface DictionaryEntry extends Entry<string, string> {
}
interface Entry<K, V> {
}
```
generated TypeScript code after change is:
``` typescript
interface DictionaryEntryUsage {
    dictionaryEntries: { [index: string]: any }[];  // better
}
```

**Comment by beleta** [comment]
Thank you Vojtěch for your help, @JsonFormat solved my problem.

**Comment by vojtechhabarta** [comment]
Released [2.19.575](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.19.575) with mentioned commit.
